src/commands/aksCompareCluster/aksCompareCluster.ts

```diff
@@ -0,0 +1,123 @@
+import { QuickPickItem } from 'vscode';
+import { createQuickPickStep, runMultiStepInput } from '../utils/multiStepHelper';
+import { IActionContext } from "@microsoft/vscode-azext-utils";
+import { getAksClusterSubscriptionItem, getContainerClient, getResourceManagementClient, } from '../utils/clusters';
+import { failed } from '../utils/errorable';
+import * as vscode from 'vscode';
+import * as k8s from 'vscode-kubernetes-tools-api';
+import SubscriptionTreeItem from '../../tree/subscriptionTreeItem';
+import { getExtension } from '../utils/host';
+import { Dictionary } from '../utils/dictionary';
+import * as tmpfile from "../utils/tempfile";
+import { longRunning } from "../utils/host";
+
+interface State {
+    clusterGroupCompareFrom: ClusterNameItem;
+    clusterGroupCompareWith: ClusterNameItem;
+    clustername: string;
+    subid: string | undefined;
+}
+
+interface ClusterNameItem extends QuickPickItem {
+    name: string;
+}
+
+/**
+ * A multi-step input using window.createQuickPick() and window.createInputBox().
+ * 
+ * This first part uses the helper class `MultiStepInput` that wraps the API for the multi-step case.
+ */
+export default async function aksCompareCluster(
+    _context: IActionContext,
+    target: unknown
+): Promise<void> {
+    const cloudExplorer = await k8s.extension.cloudExplorer.v1;
+
+    const cluster = getAksClusterSubscriptionItem(target, cloudExplorer);
+    if (failed(cluster)) {
+        vscode.window.showErrorMessage(cluster.error);
+        return;
+    }
+    const subscriptionTreeItem = <SubscriptionTreeItem>cluster.result;
```

Comment:
I've not noticed this before, and I know there's a lot of code that already uses this, so we probably shouldn't change it here...but ideally we wouldn't need a cast here. We can revisit this later.

Reply:
Sounds awesome, thank you! ☕️🥷 I have made rest of the recommended changes.